edgepi-pwm.js

```diff
@@ -0,0 +1,110 @@
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
+
+  function PWMNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
+    const pwmPin = rpc.PWMPins[config.pwmPin];
+
+    setInitialConfigs(config, node)
+      .then((pwm) => {
+        node.on("input", async function (msg, send, done) {
+          node.status({ fill: "green", shape: "dot", text: "input received" });
+          try {
+            await updateConfigsOnInput(msg, pwm, config);
+            msg = {
+              payload: await pwm.getDutyCycle(pwmPin),
+              pwmPin: pwmPin,
+              frequency: await pwm.getFrequency(pwmPin),
+              polarity: await pwm.getPolarity(pwmPin),
+              enabled: await pwm.getEnabled(pwmPin),
+            };
+          } catch (error) {
+            console.error(error);
+            msg.payload = error;
+          }
+          send(msg);
+          if (done) done();
```

Comment:
just for my understanding but does this mean sometimes there will be no `done` function? also what might the `done` function do and is it a critical step?

Reply:
done() is a callback function that indicates to node-red that the node's execution is finished. From what I understand, the if-check is just for backwards compatibility (for node-red, some earlier versions don't use it).

Reply:
ah I see, sounds good